QSPIFBlockDevice.cpp

```diff
@@ -827,79 +826,76 @@ int QSPIFBlockDevice::_sfdp_set_quad_enabled(uint8_t *basic_param_table_ptr)
 
         case 1:
         case 4:
-            status_reg_setup[0] = 0;
-            status_reg_setup[1] = 0x02;  //Bit 1 of Status Reg 2
-            tr_debug("DEBUG: Setting QE Bit, Bit 1 of Status Reg 2");
+            status_reg_setup[1] = (0x1 << 1);  //Bit 1 of Status Reg 2
+            tr_debug("Setting QE Bit, Bit 1 of Status Reg 2");
             break;
 
         case 2:
-            status_reg_setup[0] = 0x40; // Bit 6 of Status Reg 1
-            status_reg_setup[1] = 0;
+            status_reg_setup[0] = (0x1 << 6); // Bit 6 of Status Reg 1
             sr_write_size = 1;
-            sr_read_size = 1;
-            tr_debug("DEBUG: Setting QE Bit, Bit 6 of Status Reg 1");
+            tr_debug("Setting QE Bit, Bit 6 of Status Reg 1");
             break;
 
         case 3:
-            status_reg_setup[0] = 0x80; // Bit 7 of Status Reg 1
-            status_reg_setup[1] = 0;
+            status_reg_setup[0] = (0x1 << 7); // Bit 7 of Status Reg 1
             sr_write_size = 1;
-            sr_read_size = 1;
             write_register_inst = 0x3E;
             read_register_inst = 0x3F;
-            tr_debug("DEBUG: Setting QE Bit, Bit 7 of Status Reg 1");
+            tr_debug("Setting QE Bit, Bit 7 of Status Reg 1");
             break;
         case 5:
-            status_reg_setup[0] = 0;
-            status_reg_setup[1] = 0x02; //Bit 1 of status Reg 2
+            status_reg_setup[1] = (0x1 << 1); // Bit 1 of status Reg 2
             read_register_inst = 0x35;
             sr_read_size = 1;
-            tr_debug("DEBUG: Setting QE Bit, Bit 1 of Status Reg 2 -special read command");
+            tr_debug("Setting QE Bit, Bit 1 of Status Reg 2 -special read command");
             break;
         default:
-            tr_warning("WARNING: _setQuadEnable - Unsuported QER configuration");
+            tr_warning("_setQuadEnable - Unsuported QER configuration");
             break;
-
-
     }
 
+    // Configure  BUS Mode to 1_1_1 for all commands other than Read
+    _qspi_configure_format(QSPI_CFG_BUS_SINGLE, QSPI_CFG_BUS_SINGLE, QSPI_CFG_ADDR_SIZE_24, QSPI_CFG_BUS_SINGLE,
+                           QSPI_CFG_ALT_SIZE_8, QSPI_CFG_BUS_SINGLE, 0);
+
     // Read Status Register
     if (QSPI_STATUS_OK == _qspi_send_general_command(read_register_inst, QSPI_NO_ADDRESS_COMMAND, NULL, 0,
-            (char *)status_reg,
+            status_reg,
             sr_read_size) ) {  // store received values in status_value
-        tr_debug("DEBUG: Reading Status Register Success: value = 0x%x\n", (int)status_reg[0]);
+        tr_debug("Reading Status Register Success: value = 0x%x", (int)status_reg[0]);
     } else {
-        tr_error("ERROR: Reading Status Register failed");
+        tr_error("Reading Status Register failed");
         return -1;
     }
 
     // Set Bits for Quad Enable
-    status_reg[0] |= status_reg_setup[0];
-    status_reg[1] |= status_reg_setup[1];
+    for (int i = 0; i < QSPI_MAX_STATUS_REGISTER_SIZE; i++) {
```

Comment:
IMO loop just adds CPU overhead and does not make the code clearer for this case of 2 register bytes.

Reply:
Status register can be 1 byte as well for different NAND types, hence this update so that single MAX define change should work

Reply:
understood, thanks